src/dos/program_mouse.cpp

```diff
@@ -0,0 +1,163 @@
+	// 1 (30Hz), 2 (50Hz), 3 (100Hz), or 4 (200Hz).
+
+	// TODO: Find out what /Mn option of the Microsoft driver exactly does.
+
+	// TODO: Implement missing mouse driver functions and the relevant
+	// configuration switches: rotation angle (/Or, values 0-359), clickload
+	// (/KC to enable, /K to disable), button mapping (/KPnSm, values 1-3),
+	// cursor delay (/Nn, values 0-255), ballistic curve selection (/Pn,
+	// values 1-4), double speed threshold setting (/Dn).
+
+	// TODO: Implementing original options to change the driver language
+	// value: /LD (German), /LE (Spanish), /LF (French), /LI (Italian),
+	// /LK (Korean), /LJ (Japanese), /LNL (Dutch), /LP (Portuguese),
+	// /LS (Swedish), /LSF (Finnish)
+
+	// The quiet mode should not inhibit error messages - checked with
+	// Microsoft Mouse Driver v9.01
+	const bool has_option_quiet = cmd->FindExistRemoveAll("/q");
+
+	// Microsoft option to load the driver, currently the only supported
+	// action
+	cmd->FindExistRemoveAll("on");
+
+	// Check if user requested to proble for mouse port
+	bool warn_port_selection = cmd->FindExistRemoveAll("/f");
+	// Check if user requested PS/2 (/z), bus (/b), inport (/i1, /i2), or
+	// serial (/c1 - /c4) mouse port
+	for (const auto option : { "/z", "/b", "/i1", "/i2",
+	                           "/c1", "/c2", "/c3", "/c4" }) {
+		if (cmd->FindExistRemoveAll(option)) {
+			warn_port_selection = true;
+		}		
+	}
+	if (warn_port_selection) {
+		WriteOut(MSG_Get("PROGRAM_MOUSE_PORT_SELECTION"));	
+	}
+
+	// This option disables hardware mouse cursor on some cards
+	if (cmd->FindExistRemoveAll("/y")) {
+		WriteOut(MSG_Get("PROGRAM_MOUSE_HARDWARE_CURSOR"));
+	}
+
+	const bool has_option_low_memory = cmd->FindExistRemoveAll("/e");
+
+	// Check for unsupported/errorneous arguments
+	const auto params = cmd->GetArguments();
+	if (!params.empty()) {
+		WriteOut(MSG_Get("SHELL_SYNTAX_ERROR"));
+		return;
+	}
+
+	// Check if Windows is running
+	if (WINDOWS_IsStarted()) {
+		WriteOut(MSG_Get("SHELL_CANT_RUN_UNDER_WINDOWS"));
+		return;
+	}
+
+	// Check if our simulated mouse driver is started
+	if (MOUSEDOS_IsDriverStarted()) {
+		WriteOut(MSG_Get("PROGRAM_MOUSE_ALREADY_INSTALLED"));
+		return;
+	}
+
+	// Check if 3rd party mouse driver is started
+	reg_ax = 0x00;
+	CALLBACK_RunRealInt(0x33);
+	if (reg_ax == 0xFFFF) {
+		WriteOut(MSG_Get("PROGRAM_MOUSE_ALREADY_INSTALLED"));
+		return;
+	}
+
+	// Try to start the driver
+	if (!MOUSEDOS_StartDriver(has_option_low_memory)) {
+		WriteOut(MSG_Get("PROGRAM_MOUSE_COULD_NOT_INSTALL"));
+		return;
+	}
+
+	if (!has_option_quiet) {
+		WriteOut(MSG_Get("PROGRAM_MOUSE_INSTALLED"));
+	}
+}
+
+void MOUSE::AddMessages()
+{
+	MSG_Add("PROGRAM_MOUSE_HELP_LONG",
+	        "Load the built-in mouse driver.\n"
+	        "\n"
+	        "Usage:\n"
+	        "  [color=light-green]mouse[reset] [on] [/e] [/q]\n"
+	        "\n"
+	        "Parameters:\n"
+	        "  on    load the driver (default action)\n"
```

Comment:
Maybe just "load driver" for consistency with the next item just below this.

Reply:
OK, changed.